src/operator/timeout.ts

```diff
@@ -20,7 +20,8 @@ export function timeout<T>(this: Observable<T>, due: number | Date,
                            scheduler: Scheduler = async): Observable<T> {
   const absoluteTimeout = isDate(due);
   const waitFor = absoluteTimeout ? (+due - scheduler.now()) : Math.abs(<number>due);
-  const error = errorToSend || new TimeoutError();
+  const error = errorToSend instanceof Error ? errorToSend : new TimeoutError(errorToSend);
```

Comment:
> Indeed, Observable#timeout() isn't much more than a specialization of Observable.throw()

They are mostly unrelated. `Observable.throw` is for creating a lazy observable which only does one thing: call `observer.error()` with your provided error value.

`.timeout(x)` is an operator which takes the source observable and mirrors it, until the timeout ms have been exceeded and then errors with a timeout exception.

So it *has* to throw _some error_ for the timeout meaning three options: Require devs provide an exception class to throw, use our own internal exception class, or a combination of the two. We used to support a combination of the two, but we're not happy with the way it worked, and we don't want to let our bikeshedding hold back the v5-final release, so we made the call to remove the ability to customize the exception, and instead just us our own consistently.

There was the second point of discussion which is creating the error object at the operator callsite so we have better stack traces in the event of a timeout, that is seemingly unrelated to whether or not we accept a custom error class or not cause the argument still stands for either option. We discussed the pros vs cons of creating that instance at the callsite vs. lazy, and generally came to the consensus that the developer experience outweighs the (in practice) _very_ small hit of allocating the object we may or not not use.

I can totally listen to reason, but I'd need more guidance than saying we did it for no good reason.